tests/update_workflow.go

```diff
@@ -3874,16 +3904,21 @@ func (s *FunctionalSuite) TestUpdateWorkflow_FirstNormalWorkflowTaskUpdateLost_B
 	}
 	go updateWorkflowFn()
 
-	// Process update in workflow. Update won't be found on server due to shard reload and server will fail WT.
-	_, err := poller.PollAndProcessWorkflowTask()
-	s.Error(err)
-	s.IsType(&serviceerror.InvalidArgument{}, err, "workflow task failure must be an InvalidArgument error")
-	s.ErrorContains(err, fmt.Sprintf("update %s wasn't found", tv.UpdateID("1")))
+	// Process update in workflow. Update won't be found on server but will be resurrected from acceptance message and completed.
+	res, err := poller.PollAndProcessWorkflowTask(WithForceNewWorkflowTask)
+	s.NoError(err)
+	s.NotNil(res)
 
+	updateResp := res.NewTask
 	<-updateResultCh
```

Comment:
Is there something higher up the stack causing the test to timeout, or does this need to be raced against a timeout?

(I believe I may be guilty of having introduced such channel reads into the test suite without racing against timeouts)

Reply:
Entire test can complete before timeout expires and asserts in go routine are made.